core/dao.js

```diff
@@ -705,10 +705,10 @@ CLASS({
       swiftCode: function() {/*
         let future = Future()
         let removeSink = ClosureSink(args: [
-          "putFn": FoamFunction(fn: { (args) -> AnyObject? in
+          "putFn": FoamFunction(fn: { (args: AnyObject?...) -> AnyObject? in
```

Comment:
There was regression in Swift 4 that is not yet fixed:
https://bugs.swift.org/browse/SR-6030 & https://bugs.swift.org/browse/SR-2443

Because of that variadic parameters require explicit type declaration in a closure, even if it's otherwise possible to infer it from a context.